Source/WebCore/platform/graphics/gstreamer/MediaPlayerPrivateGStreamerMSE.cpp

```diff
@@ -701,6 +704,10 @@ void MediaPlayerPrivateGStreamerMSE::updateStates()
                     m_readyState = MediaPlayer::HaveEnoughData;
                     LOG_MEDIA_MESSAGE("m_readyState=%s", dumpReadyState(m_readyState));
                     m_networkState = m_downloadFinished ? MediaPlayer::Idle : MediaPlayer::Loading;
+                } else if (m_mediaSource) {
```

Comment:
Yes, it's ok. I've tested the changes you're proposing and both tests and Youtube TV keep working.

The updateStates() code in the MSE player was based on the code from the regular player. The m_buffering field is only set from events coming from a buffering element which is only present in the regular pipeline (not in MSE). Therefore, the m_buffering branch is actually dead code now and can safely be removed like you propose.

In the same way, after having a look to the code and adding some log messages to double check, it looks like the m_downloadFinished field is always set to false in the MSE player, so the code could remain like this:

```
        case GST_STATE_PAUSED:
        case GST_STATE_PLAYING:
            if (seeking()) {
                m_readyState = MediaPlayer::HaveMetadata;
                // TODO: NetworkState?
            } else if (m_mediaSource) {
                m_mediaSource->monitorSourceBuffers();
                m_networkState = MediaPlayer::Loading;
            }
            LOG_MEDIA_MESSAGE("m_readyState=%s", dumpReadyState(m_readyState));
            break;
```